File: src/wr_logic_teleop_ds/src/Watchdog.cpp
#include "Watchdog.hpp"
#include <chrono>

Watchdog::Watchdog(double timeout){
    this->timeout = timeout;
    this->pet();
}

void Watchdog::pet(){
    this->lastTime = std::chrono::system_clock::now();
}

bool Watchdog::isMad(){
    return std::chrono::duration<double>(std::chrono::system_clock::now() - this->lastTime).count() >= this->timeout;
}

File: src/wr_logic_teleop_arm/nodes/watchdog.py
import time

class Watchdog:
    def __init__(self, timeout: float):
        self._timeout = timeout
        self.pet()

    def pet(self):
        self._lastTime = time.clock_gettime(0)

    def isMad(self):
        return time.clock_gettime(0) - self._lastTime >= self._timeout

